### ISSUE TITLE
Strange issue, I have just trained for 5 epoches, the miou is decreasing and I stopped it

### ISSUE THREAD

**Issue by XUYUNYUN666** [jfzhang95/pytorch-deeplab-xception#135]
Namespace(backbone='resnet', base_size=513, batch_size=4, checkname='deeplab-resnet', crop_size=513, cuda=True, dataset='cityscapes', epochs=20, eval_interval=1, freeze_bn=False, ft=False, gpu_ids=[0], loss_type='ce', lr=0.01, lr_scheduler='poly', momentum=0.9, nesterov=False, no_cuda=False, no_val=False, out_stride=16, resume=None, seed=1, start_epoch=0, sync_bn=False, test_batch_size=4, use_balanced_weights=False, use_sbd=True, weight_decay=0.0005, workers=4)
Found 1000 train images
Found 296 val images
Downloading: "https://download.pytorch.org/models/resnet101-5d3b4d8f.pth" to C:\Users\Xuyunyun/.cache\torch\checkpoints\resnet101-5d3b4d8f.pth
100%|██████████| 170M/170M [01:41<00:00, 1.77MB/s]
Using poly LR Scheduler!
  0%|          | 0/250 [00:00<?, ?it/s]Starting Epoch: 0
Total Epoches: 20

=>Epoches 0, learning rate = 0.0100,                 previous best = 0.0000
C:\ProgramData\Anaconda3\envs\pytorch-gpu\lib\site-packages\torch\nn\_reduction.py:43: UserWarning: size_average and reduce args will be deprecated, please use reduction='mean' instead.
  warnings.warn(warning.format(ret))
Train loss: 0.092: 100%|██████████| 250/250 [06:21<00:00,  1.53s/it]
[Epoch: 0, numImages:  1000]
:   0%|          | 0/74 [00:00<?, ?it/s]Loss: 23.110
Test loss: 0.118: 100%|██████████| 74/74 [00:52<00:00,  1.47it/s]Validation:
[Epoch: 0, numImages:   296]
Acc:0.8675745099773006, Acc_class:0.4810683262735606, mIoU:0.41107297819611816, fwIoU: 0.7912389436073557
Loss: 8.762
Test loss: 0.118: 100%|██████████| 74/74 [00:52<00:00,  1.42it/s]
  0%|          | 0/250 [00:00<?, ?it/s]
=>Epoches 1, learning rate = 0.0095,                 previous best = 0.4111
Train loss: 0.068: 100%|██████████| 250/250 [06:29<00:00,  1.56s/it]
[Epoch: 1, numImages:  1000]
:   0%|          | 0/74 [00:00<?, ?it/s]Loss: 16.979
Test loss: 0.217: 100%|██████████| 74/74 [00:51<00:00,  1.53it/s]Validation:
Test loss: 0.217: 100%|██████████| 74/74 [00:52<00:00,  1.42it/s]
[Epoch: 1, numImages:   296]
  0%|          | 0/250 [00:00<?, ?it/s]Acc:0.8293775205389036, Acc_class:0.5018260896262577, mIoU:0.40737300417162364, fwIoU: 0.7217043209719889
Loss: 16.085

=>Epoches 2, learning rate = 0.0091,                 previous best = 0.4111
Train loss: 0.060: 100%|██████████| 250/250 [06:32<00:00,  1.57s/it]
[Epoch: 2, numImages:  1000]
:   0%|          | 0/74 [00:00<?, ?it/s]Loss: 15.039
Test loss: 0.331: 100%|██████████| 74/74 [00:52<00:00,  1.53it/s]Validation:
Test loss: 0.331: 100%|██████████| 74/74 [00:52<00:00,  1.40it/s]
[Epoch: 2, numImages:   296]
  0%|          | 0/250 [00:00<?, ?it/s]Acc:0.7632570756865411, Acc_class:0.4936742116673538, mIoU:0.38041394196087513, fwIoU: 0.6364431186645987
Loss: 24.523

=>Epoches 3, learning rate = 0.0086,                 previous best = 0.4111
Train loss: 0.054: 100%|██████████| 250/250 [06:31<00:00,  1.57s/it]
:   0%|          | 0/74 [00:00<?, ?it/s][Epoch: 3, numImages:  1000]
Loss: 13.586
Test loss: 0.448: 100%|██████████| 74/74 [00:52<00:00,  1.41it/s]
Validation:
[Epoch: 3, numImages:   296]
Acc:0.776739625641852, Acc_class:0.4754543121798434, mIoU:0.3711352410521569, fwIoU: 0.6491044481051552
Loss: 33.168
  0%|          | 0/250 [00:00<?, ?it/s]
=>Epoches 4, learning rate = 0.0082,                 previous best = 0.4111
Train loss: 0.052: 100%|██████████| 250/250 [06:30<00:00,  1.56s/it]
[Epoch: 4, numImages:  1000]
:   0%|          | 0/74 [00:00<?, ?it/s]Loss: 13.083
Test loss: 0.580: 100%|██████████| 74/74 [00:53<00:00,  1.39it/s]
Validation:
[Epoch: 4, numImages:   296]
Acc:0.7439612974008173, Acc_class:0.47899480905874386, mIoU:0.35453582991815424, fwIoU: 0.6075277612551088
Loss: 42.884
  0%|          | 0/250 [00:00<?, ?it/s]
=>Epoches 5, learning rate = 0.0077,                 previous best = 0.4111
Train loss: 0.052:  72%|███████▏  | 181/250 [04:44<01:47,  1.55s/it]

**Comment by vazyavazya** [comment]
May be initial learning rate = 0.0100 is too high? Any way I have the same issue with xception and resnet backbone with different initial learning rate.

**Comment by vazyavazya** [comment]
Also there is problem with freezeng batchnorms, due to it becomes freezed only in init of Deeplab, try next code in initialization of deeplab.
```
...
        self.bn_freezed = freeze_bn
        if freeze_bn:
            self.freeze_bn()

    def forward(self, input):
        if self.bn_freezed:
            self.freeze_bn()
        x, low_level_feat = self.backbone(input)
        x = self.aspp(x)
        x = self.decoder(x, low_level_feat)
        x = F.interpolate(x, size=input.size()[2:], mode='bilinear', align_corners=True)

        return x
...
```

**Comment by jfzhang95** [comment]
I have updated the code, please try again.
